### PR TITLE
Added a new popGrapheme function to std.uni

### PR DESCRIPTION
**Pull request by dukc**
Since it's Hackathon today, even if I'm not there I decided to do something in similar spirit.

@atilaneves new symbol canditate for phobos, meaning it needs your approval!

### CODE REVIEW

**Comment by dukc** [comment]
ping @atilaneves

File: changelog/pop-grapheme.dd [hunk 1]
```diff
@@ -0,0 +1,38 @@
+Added popGrapheme function to std.uni.
+
+The new function is a cross between the existing $(REF graphemeStride, std,
+uni) and $(REF decodeGrapheme, std, uni) functions. The new function both
+supports `@safe pure nothrow @nogc` like `graphemeStride` does as long as you
+don't rely on autodecoding (side node: `@nogc` support for `graphemeStride`
```
Comment (atilaneves) [inline changelog/pop-grapheme.dd:6]: This side note sounds like it should be its own PR.

Comment (dukc) [reply changelog/pop-grapheme.dd:6]: It [is](https://github.com/dlang/phobos/pull/9050). I felt an added `@nogc` compability to an existing funtion alone wasn't worth a changelog entry so didn't add one in that PR. But since I'm writing here that `graphemeStride` is `@nogc` anyway, which until very recently wasn't the case, I feel it's useful to mention it. As in: "this note applies only with the latest compiler/Phobos".

**Comment by atilaneves** [comment]
I don't know how useful this is so I would like someone that knows/cares about unicode more than me opines. Not sure who. @rikkimax ?

**Comment by rikkimax** [comment]
> I don't know how useful this is so I would like someone that knows/cares about unicode more than me opines. Not sure who. @rikkimax ?

It allows you to get the grapheme length extracted, so you can slice and dice your string without allocation.

Kinda like ``graphemeStride`` except it pops it from the input like ``parse`` does.

It is one of those things that if a unicode library is being used you'll end up having a hundred variations of. I see nothing wrong with wanting functions like this.

**Comment by dukc** [comment]
> I don't know how useful this is so I would like someone that knows/cares about unicode more than me opines. Not sure who.

Maybe @John-Colvin ? I think he was supposed to take charge, at one point at least, of my Unicode-related project at Symmetry.

**Comment by atilaneves** [comment]
> > I don't know how useful this is so I would like someone that knows/cares about unicode more than me opines. Not sure who. @rikkimax ?
> 
> It allows you to get the grapheme length extracted, so you can slice and dice your string without allocation.
> 
> Kinda like `graphemeStride` except it pops it from the input like `parse` does.
> 
> It is one of those things that if a unicode library is being used you'll end up having a hundred variations of. I see nothing wrong with wanting functions like this.

So you're saying we should merge it?

**Comment by rikkimax** [comment]
> > > I don't know how useful this is so I would like someone that knows/cares about unicode more than me opines. Not sure who. @rikkimax ?
> > 
> > 
> > It allows you to get the grapheme length extracted, so you can slice and dice your string without allocation.
> > Kinda like `graphemeStride` except it pops it from the input like `parse` does.
> > It is one of those things that if a unicode library is being used you'll end up having a hundred variations of. I see nothing wrong with wanting functions like this.
> 
> So you're saying we should merge it?

I have no argument against merging. It is sound, it's a variation of something that exists and is worth having different variants for.

The reason I'm avoiding the phrase "merge it" is because of PhobosV3 vs V2. I don't know how you and @LightBender want to handle it.

**Comment by LightBender** [comment]
I'm good with merging this.

**Review by LightBender — APPROVED** [review]

**Comment by dukc** [comment]
> Need to get it passing the build though.

Fails at some of the platforms but passes on others, and I'm not doing anything platform-specific. So probably spurious. I'll attempt restarting the tests.

**Comment by dukc** [comment]
Windows 64bit test is now passing, Mac tests are still failing. Seems to be the same issue in #9056 . I wonder what's going on there. @RazvanN7 do you know?

**Comment by RazvanN7** [comment]
It seems that the Mac machines are broken. Most likely this PR is not at fault.

**Comment by dukc** [comment]
Well Mac machines don't apparently have the "required" labes so probably should just ignore them.

@atilaneves good to go now?

**Comment by LightBender** [comment]
I'm going to merge this as there has been no dissent to date.
